### DISCUSSION TITLE
Parsing Returns the Wrong Type

### DISCUSSION THREAD

**Discussion by ajs11174** [colinhacks/zod#2263] [Q&A] [answered]
The type extracted from parse and safe parse are optional on a basic ```z.object({})```.

I put together an codesandbox to demonstrate the issue based on the 2nd basic usage example.

[https://github.com/colinhacks/zod#basic-usage](https://github.com/colinhacks/zod#basic-usage)
[https://codesandbox.io/p/sandbox/angry-shadow-t3ot3s](https://codesandbox.io/p/sandbox/angry-shadow-t3ot3s?file=%2Fsrc%2Findex.ts)

Not sure if this is a bug or if I have something misconfigured. 

btw: the library is awesome

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#requirements
> You must enable `strict` mode in your `tsconfig.json`. This is a best practice for all TypeScript projects.

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by ajs11174** [reply]
Thanks! Was moving a project over to a monorepo and missed strict mode.

**Reply by vitormarkis** [reply]
it helped !
